pkg/services/ngalert/state/state.go

```diff
@@ -228,11 +238,14 @@ func resultError(state *State, rule *models.AlertRule, result eval.Result, logge
 		if state.State == eval.Error {
 			logger.Debug("Keeping state", "state", state.State)
 			state.Maintain(rule.IntervalSeconds, result.EvaluatedAt)
+		} else if state.State == eval.Pending && state.Error != nil {
```

Comment:
This case is the opposite of the one above.

Imagine that the current state of an alert is Pending (meaning there have been alerting states and the for has not been reached) [current state is state.State == eval.Pending and state.Error == nil]

If for some reason the datasource is not reachable and it is immediately erroring we do NOT want the flow to enter this case statement. We want to skip it and go to the default branching instead as it is the first time it is firing. The next state will be state.State == eval.Pending and state.Error != nil

Reply:
It does not apply anymore. This behavior has changed.